src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2255 @@
+            comp.VerifyDiagnostics(
+                // (8,9): error CS8802: Init-only member 'C.Property' can only be assigned from a constructor, object initialization or 'with' expression of that type.
+                //         Property++; // 1
+                Diagnostic(ErrorCode.ERR_AssignmentInitOnly, "Property").WithArguments("C.Property").WithLocation(8, 9)
+                );
+        }
+
+        [Fact]
+        public void RefProperty()
+        {
+            string source = @"
+public class C
+{
+    ref int Property1 { get; init; }
+    ref int Property3 { init; }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (4,13): error CS8145: Auto-implemented properties cannot return by reference
+                //     ref int Property1 { get; init; }
+                Diagnostic(ErrorCode.ERR_AutoPropertyCannotBeRefReturning, "Property1").WithArguments("C.Property1").WithLocation(4, 13),
+                // (4,30): error CS8147: Properties which return by reference cannot have set accessors
+                //     ref int Property1 { get; init; }
+                Diagnostic(ErrorCode.ERR_RefPropertyCannotHaveSetAccessor, "init").WithArguments("C.Property1.set").WithLocation(4, 30),
+                // (5,13): error CS8146: Properties which return by reference must have a get accessor
+                //     ref int Property3 { init; }
+                Diagnostic(ErrorCode.ERR_RefPropertyMustHaveGetAccessor, "Property3").WithArguments("C.Property3").WithLocation(5, 13)
+                );
+        }
+
+        [Fact]
+        public void VerifyPESymbols_Property()
+        {
+            string source = @"
+public class C
+{
+    public string Property { get; init; }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition },
+                parseOptions: TestOptions.RegularPreview,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All));
+            comp.VerifyDiagnostics();
+
+            if (ExecutionConditionUtil.IsCoreClr)
+            {
+                CompileAndVerify(comp, sourceSymbolValidator: symbolValidator, symbolValidator: symbolValidator);
+            }
+            else
+            {
+                // PE verification fails:  [ : C::set_Property] Cannot change initonly field outside its .ctor.
+                CompileAndVerify(comp, sourceSymbolValidator: symbolValidator, symbolValidator: symbolValidator, verify: Verification.Fails);
+            }
+
+            void symbolValidator(ModuleSymbol m)
+            {
+                bool isSource = !(m is PEModuleSymbol);
+                var c = (NamedTypeSymbol)m.GlobalNamespace.GetMember("C");
+
+                var property = (PropertySymbol)c.GetMembers("Property").Single();
```

Comment:
>var property = (PropertySymbol)c.GetMembers("Property").Single(); [](start = 16, length = 65)

Please assert modifiers on the property. We should make sure that we don't emit them on the property signature